src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -0,0 +1,570 @@
+        if location is None:
+            dynamic_completion = True
+        # In case the user does not specify the parameter and it meets the conditions of automatic completion,
+        # necessary information is dynamically completed.
+        if dynamic_completion:
+            location = _get_rg_location(
+                self.cmd.cli_ctx, self.get_resource_group_name()
+            )
+            # add to intermediate
+            self.set_intermediate(
+                parameter_name, location, overwrite_exists=True
+            )
+
+        # this parameter does not need validation
+        return location
+
+    def get_kubernetes_version(
+        self,
+        enable_validation: bool = False,
+        force_update: bool = False,
+        **kwargs
+    ):
+        # Note: This parameter will not be decorated into the `mc` object.
+        # read the original value passed by the command
+        kubernetes_version = getattr(self.raw_param, "kubernetes_version")
+
+        # this parameter does not need dynamic completion
+        # this parameter does not need validation
+        return kubernetes_version
+
+    def get_no_ssh_key(
+        self,
+        enable_validation: bool = False,
+        force_update: bool = False,
+        **kwargs
+    ):
+        # read the original value passed by the command
+        no_ssh_key = getattr(self.raw_param, "no_ssh_key")
+
+        # this parameter does not need dynamic completion
+
+        # validation
+        if enable_validation:
+            _validate_ssh_key(
+                no_ssh_key=no_ssh_key, ssh_key_value=self.get_ssh_key_value()
+            )
+        return no_ssh_key
+
+    def get_vm_set_type(
+        self,
+        enable_validation: bool = False,
+        force_update: bool = False,
+        **kwargs
+    ):
+        parameter_name = "vm_set_type"
+
+        # read the original value passed by the command
+        raw_value = getattr(self.raw_param, parameter_name)
+        # try to read from intermediates
+        intermediate = self.get_intermediate(parameter_name, None)
+        # try to read the property value corresponding to the parameter from the `mc` object
+        mc_property = None
+        if self.mc and self.mc.agent_pool_profiles:
+            agent_pool_profile = safe_list_get(
+                self.mc.agent_pool_profiles, 0, None
+            )
+            if agent_pool_profile:
+                mc_property = agent_pool_profile.type
+
+        # set defautl value
+        if mc_property is not None:
+            vm_set_type = mc_property
+            # clean up intermediate if `mc` has been decorated
+            self.remove_intermediate(parameter_name)
+        elif intermediate is not None:
+            vm_set_type = intermediate
+        else:
+            vm_set_type = raw_value
+
+        # enable dynamic completion if the `force_update` parameter is specified when calling this getter
+        dynamic_completion = force_update
```

Comment:
suggest removing the force_update.

Reply:
Good point, removed in commit [7919912](https://github.com/Azure/azure-cli/pull/19259/commits/79199128f0819b2110088c6f8d39f0aab8446bfe).